internal/blocksync/reactor.go

```diff
@@ -504,7 +507,7 @@ func (bcR *Reactor) isMissingExtension(state sm.State, blocksSynced uint64) bool
 }
 
 func (bcR *Reactor) isCaughtUp(state sm.State, blocksSynced uint64, stateSynced bool) bool {
-	if isCaughtUp, height, _ := bcR.pool.IsCaughtUp(); isCaughtUp {
+	if isCaughtUp, height, _ := bcR.pool.IsCaughtUp(state, bcR.myAddr); isCaughtUp {
```

Comment:
We can add the addition condition here, actually. We have the state, etc.

Reply:
I don't understand your comment

Reply:
We can check this condition here, at the reactor, before checking the pool.

Reply:
Oh I see, let me explore that possibility

Reply:
Man... you deserve a bonus for this comment! Everything is so much easier with the check outside the pool! 🚀